src/main/java/frc/trigon/robot/subsystems/swerve/SwerveConstants.java

```diff
@@ -33,36 +37,20 @@ public class SwerveConstants {
             FRONT_RIGHT_ID = 2,
             REAR_LEFT_ID = 3,
             REAR_RIGHT_ID = 4;
-    private static final double
-            FRONT_LEFT_WHEEL_DIAMETER_METERS = RobotHardwareStats.isSimulation() ? 0.1016 : 0.049274 * 2,
-            FRONT_RIGHT_WHEEL_DIAMETER_METERS = RobotHardwareStats.isSimulation() ? 0.1016 : 0.049274 * 2,
-            REAR_LEFT_WHEEL_DIAMETER_METERS = RobotHardwareStats.isSimulation() ? 0.1016 : 0.049274 * 2,
-            REAR_RIGHT_WHEEL_DIAMETER_METERS = RobotHardwareStats.isSimulation() ? 0.1016 : 0.049274 * 2;
+    public static final double WHEEL_DIAMETER_METERS = ROBOT_CONFIG.moduleConfig.wheelRadiusMeters * 2;
     static final SwerveModule[] SWERVE_MODULES = {
-            new SwerveModule(FRONT_LEFT_ID, FRONT_LEFT_STEER_ENCODER_OFFSET, FRONT_LEFT_WHEEL_DIAMETER_METERS),
-            new SwerveModule(FRONT_RIGHT_ID, FRONT_RIGHT_STEER_ENCODER_OFFSET, FRONT_RIGHT_WHEEL_DIAMETER_METERS),
-            new SwerveModule(REAR_LEFT_ID, REAR_LEFT_STEER_ENCODER_OFFSET, REAR_LEFT_WHEEL_DIAMETER_METERS),
-            new SwerveModule(REAR_RIGHT_ID, REAR_RIGHT_STEER_ENCODER_OFFSET, REAR_RIGHT_WHEEL_DIAMETER_METERS)
+            new SwerveModule(FRONT_LEFT_ID, FRONT_LEFT_STEER_ENCODER_OFFSET),
+            new SwerveModule(FRONT_RIGHT_ID, FRONT_RIGHT_STEER_ENCODER_OFFSET),
+            new SwerveModule(REAR_LEFT_ID, REAR_LEFT_STEER_ENCODER_OFFSET),
+            new SwerveModule(REAR_RIGHT_ID, REAR_RIGHT_STEER_ENCODER_OFFSET)
     };
 
     private static final DoubleSupplier SIMULATION_YAW_VELOCITY_SUPPLIER = () -> RobotContainer.SWERVE.getSelfRelativeVelocity().omegaRadiansPerSecond;
-
-    private static final double
-            FRONT_MODULE_X_DISTANCE_FROM_CENTER = 0.25,
-            FRONT_MODULE_Y_DISTANCE_FROM_CENTER = 0.25,
-            REAR_MODULE_X_DISTANCE_FROM_CENTER = 0.25,
-            REAR_MODULE_Y_DISTANCE_FROM_CENTER = 0.25;
-    public static final Translation2d[] MODULE_LOCATIONS = {
-            new Translation2d(FRONT_MODULE_X_DISTANCE_FROM_CENTER, FRONT_MODULE_Y_DISTANCE_FROM_CENTER),
-            new Translation2d(FRONT_MODULE_X_DISTANCE_FROM_CENTER, -FRONT_MODULE_Y_DISTANCE_FROM_CENTER),
-            new Translation2d(REAR_MODULE_X_DISTANCE_FROM_CENTER, REAR_MODULE_Y_DISTANCE_FROM_CENTER),
-            new Translation2d(REAR_MODULE_X_DISTANCE_FROM_CENTER, -REAR_MODULE_Y_DISTANCE_FROM_CENTER)
-    };
-    public static final SwerveDriveKinematics KINEMATICS = new SwerveDriveKinematics(MODULE_LOCATIONS);
+    public static final SwerveDriveKinematics KINEMATICS = new SwerveDriveKinematics(ROBOT_CONFIG.moduleLocations);
     private static final double FURTHEST_MODULE_DISTANCE_FROM_CENTER = Math.hypot(
-            REAR_MODULE_X_DISTANCE_FROM_CENTER, FRONT_MODULE_Y_DISTANCE_FROM_CENTER
+            Math.max(ROBOT_CONFIG.moduleLocations[0].getX(), ROBOT_CONFIG.moduleLocations[1].getX()),
+            Math.max(ROBOT_CONFIG.moduleLocations[0].getY(), ROBOT_CONFIG.moduleLocations[3].getY())
```

Comment:
this doesn't cover all cases

Reply:
But we won't have a case where the first 2 modules aren't lined up same with the last ones. And we won't have a case where the sides aren't lined up.

Reply:
Why not? Two plausible cases.

Reply:
Keep in mind you have an array with the norms inside the config